Products/ZenMessaging/queuemessaging/publisher.py

```diff
@@ -271,7 +273,7 @@ def beforeCompletionHook(self, tx):
             if publisher:
                 msgs = self.correlateEvents(publisher.events)
                 with _getPrepublishingTimer():
-                    notify(MessagePrePublishingEvent(msgs))
+                    notify(MessagePrePublishingEvent(msgs, maintWindowChange=publisher._maintWindowChange))
```

Comment:
This presumes all events are maintenance window changes if any one was caused by a production state change.